geoportal/c2cgeoportal_geoportal/lib/functionality.py

```diff
@@ -55,13 +56,28 @@ def _get_config_functionality(name, registered, types, request, errors):
     return [r for r in result if r is not None]
 
 
-def _get_db_functionality(name, role, types, request, errors):
-    result = [
-        get_typed(name, functionality.value, types, request, errors)
-        for functionality in role.functionalities
-        if functionality.name == name
-    ]
-    return [r for r in result if r is not None]
+def _get_db_functionality(name, user: static.User, types, request, errors):
+    if types[name].get('single', False):
+        values = [
+            get_typed(name, functionality.value, types, request, errors)
+            for functionality in user.settings_role.functionalities
+            if functionality.name == name
+        ]
+        return [r for r in values if r is not None]
+
+    else:
+        functionalities = {
+            functionality.value
+            for role in user.roles
+            for functionality in role.functionalities
+            if functionality.name == name
+        }
```

Comment:
{ => [
} => ]
and the first comment was we should use set(functionalities) to have distinct values

Reply:
{...} directly creates a set ...